src/main/java/fr/sncf/osrd/railjson/parser/RailJSONParser.java

```diff
@@ -231,7 +232,12 @@ public static Infra parse(RJSInfra railJSON) throws InvalidInfraException {
                 switchesPosition.put(switchRef, position);
             }
 
-            routeGraph.makeRoute(rjsRoute.id, waypoints, tvdSections, releaseGroups, switchesPosition);
+            var entrySignal = signals.stream()
+                    .filter(x -> x.id.equals(rjsRoute.entrySignal.id))
+                    .findFirst()
+                    .orElse(null);
```

Comment:
isn't there a hashmap you can use somewhere?

Reply:
Not at this point of the program, but the ArrayList could (and probably should) be turned into a HashMap. Is there a reason not to?